Makefile

```diff
@@ -90,6 +110,80 @@ test-silent:
 benchmark-test:
 	. ./scripts/shared_env && go test -run=XX -bench=. $(shell go list ./agent/... | grep -v /vendor/)
 
+define dockerbuild
+	docker run \
+		--net none \
+		--user "$(USERID)" \
+		--volume "$(PWD)/out:/out" \
+		--volume "$(PWD):/go/src/github.com/aws/amazon-ecs-agent" \
+		--rm \
+		$(BUILDER_IMAGE) \
+		$(1)
+endef
+
+define win-cgo-dockerbuild
+	docker run --net=none \
+		--user "$(USERID)" \
+		--volume "$(PWD)/out:/out" \
+		--volume "$(PWD):/go/src/github.com/aws/amazon-ecs-agent" \
+		--env "GOOS=windows" \
+		--env "CGO_ENABLED=1" \
+		--env "CC=x86_64-w64-mingw32-gcc" \
+		--rm \
+		$(BUILDER_IMAGE) \
+		$(1)
+endef
+
+# TODO: use `go list -f` to target the test files more directly
+ALL_GO_FILES = $(shell find . -name "*.go" -print | tr "\n" " ")
```

Comment:
Can you please address this `TODO` in this PR? Also, as per earlier comments, please move these const's to the top of the file

Reply:
In short, I tried and its a little complicated. I wanted more time to think about it -- but I'd love to make another pass and clean up more of the Makefile, including this `TODO`.

Reply:
Regarding moving this constant, It seemed more standard to have makefiles in 'sections'. But I'm largely indifferent. If you still want me to move it I will.